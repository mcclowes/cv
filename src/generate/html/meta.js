const meta = (config) => {
  if (!config || !config.name) {
    return `
        <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2,shrink-to-fit=no">

        <link rel="icon" href="assets/favicon.ico" />

        <title>CV</title>
        <meta name="description" content="CV made with cv-maker">
    `;
  }

  const {
    description,
    name,
    preview,
    twitterUser,
    url,
  } = config

  return `
    <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=2,shrink-to-fit=no">

    <link rel="icon" href="assets/favicon.ico" />

    <title>${name} CV</title>
    <meta name="description" content="${description}">

    <!--  Social tags -->    
    <meta property="og:description" content="${description}">
    <meta property="og:image" content="${preview.image}">
    <meta property="og:site_name" content="The CV of ${name}">
    <meta property="og:title" content="The CV of ${name}">
    <meta property="og:url" content="${url}">

    <meta name="twitter:card" content="summary_large_image">
    <meta name="twitter:image:alt" content="${preview.text}">
    <meta name="twitter:site" content="${twitterUser}">
  `;
};

export default meta;
