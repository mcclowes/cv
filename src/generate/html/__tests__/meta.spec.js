import meta from "../meta";

const CONFIG = {
  name: "Joe Bloggs",
  description: "Enter a summary of you here.",
  preview: {
    image: "https://cv.joebloggs.com/preview.png",
    text: "Joe Bloggs CV",
  },
  url: "https://cv.joebloggs.com/",
  twitterUsername: "@joebloggs",
};

describe("meta", () => {
  it("generates correct meta html", () => {
    expect(meta(CONFIG)).toMatchSnapshot();
  });
});
