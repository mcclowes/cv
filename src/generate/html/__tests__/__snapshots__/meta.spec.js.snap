// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meta generates correct meta html 1`] = `
"
    <meta name=\\"viewport\\" content=\\"width=device-width,initial-scale=1,maximum-scale=2,shrink-to-fit=no\\">

    <link rel=\\"icon\\" href=\\"assets/favicon.ico\\" />

    <title>Joe Bloggs CV</title>
    <meta name=\\"description\\" content=\\"Enter a summary of you here.\\">

    <!--  Social tags -->    
    <meta property=\\"og:description\\" content=\\"Enter a summary of you here.\\">
    <meta property=\\"og:image\\" content=\\"https://cv.joebloggs.com/preview.png\\">
    <meta property=\\"og:site_name\\" content=\\"The CV of Joe Bloggs\\">
    <meta property=\\"og:title\\" content=\\"The CV of Joe Bloggs\\">
    <meta property=\\"og:url\\" content=\\"https://cv.joebloggs.com/\\">

    <meta name=\\"twitter:card\\" content=\\"summary_large_image\\">
    <meta name=\\"twitter:image:alt\\" content=\\"Joe Bloggs CV\\">
    <meta name=\\"twitter:site\\" content=\\"undefined\\">
  "
`;
